src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  isProcessing: number[];
+  onStatusChange: (todoId: number) => void;
+  onTitleChange: (todoId: number, newTitle: string) => void;
+  onDeleteTodo: (todoId: number) => void;
+};
+
+export const TodoList: React.FC<Props> = (props) => {
+  const {
+    todos,
+    isProcessing,
+    onStatusChange,
+    onTitleChange,
+    onDeleteTodo,
+  } = props;
+  const selectedTodoField = useRef<HTMLInputElement>(null);
+  const [selectedTodo, setSelectedTodo] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState<string>('');
+
+  const handleStartEditTitle = (id: number, title: string) => {
+    setSelectedTodo(id);
+    setNewTitle(title);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleSaveChages = (todoId: number, title: string) => {
+    setSelectedTodo(null);
+    if (newTitle === '') {
+      onDeleteTodo(todoId);
+    } else if (newTitle !== title) {
+      onTitleChange(todoId, newTitle);
+    }
+  };
+
+  const handleCancelChanges = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      setSelectedTodo(null);
+    }
+  };
```

Comment:
You could add a check - which key was pressed to save. Now the edited todo is saved if you pressed alt or clicked mouse on another field.

Reply:
But this is exactly what we need: save changes when the field loses focus (onBlur)